app/controllers/users_controller.rb

```diff
@@ -0,0 +1,9 @@
+class UsersController < ApplicationController
+  
+  def show
+    @user = User.find(params[:id])
+    @bookmarks = @user.bookmarks
+    @liked_bookmarks = @user.likes.map { |like| like.bookmark }
+  end
```

Comment:
This works great, but can we improve the performance?  

Everytime you make a query to the DB, there is extra overheard to to send a request to the db, perform the operation, and return the result to the Rails application.  Right now it could make a query on line 4 to get the user, another query on line 5 to get the `bookmarks`, then on line 6 its making another query for EVERY `like` to get the like's `bookmark`.  If a user has 10,000 likes that could be expensive! This is known is the "n+1" query problem. (disclaimer: I say "can" because Rails does a lot of optimizations and I'm simplifying how it works)

To solve this, ActiveRecord has a feature called [Eager Loading](http://guides.rubyonrails.org/active_record_querying.html#eager-loading-multiple-associations) which allows you to specify when construct the query all of the associations you want, and ActiveRecord will request all of the pertinent data from the DB in a single query.

Can you change this method to use Eager Loading for bookmarks, likes, and the likes' bookmarks?